src/evaluation/evaluation-error-handler.service.ts

```diff
@@ -0,0 +1,350 @@
+import { Evaluation } from '../entities/evaluation.entity';
+import { v4 as uuidv4 } from 'uuid';
+import { EvaluationSetHelperService } from './evaluation-set-helper.service';
+import { MailEvalService } from '../mail/mail-eval.service';
+import { ConfigService } from '@nestjs/config';
+import { Plant } from '../entities/plant.entity';
+import { EntityManager } from 'typeorm';
+import { ReportingPeriod } from '../entities/reporting-period.entity';
+import { EvalErrorParamsDTO } from '../dto/eval-error-params.dto';
+
+@Injectable()
+export class EvaluationErrorHandlerService {
+  constructor(
+    private readonly logger: Logger,
+    private readonly entityManager: EntityManager,
+    private readonly mailEvalService: MailEvalService,
+    private readonly evaluationSetHelper: EvaluationSetHelperService,
+    private readonly configService: ConfigService,
+  ) {}
+
+  async sendQueueingErrorEmail(
+    evalErrorParamsDTO: EvalErrorParamsDTO
+  ) {
+
+    let evaluation: EvaluationSet;
+    if (evalErrorParamsDTO.evaluationSetId) {
+      evaluation = await this.entityManager.findOneBy(EvaluationSet, {
+        evaluationSetIdentifier: evalErrorParamsDTO.evaluationSetId,
+      });
+    }
+
+    if (!evaluation) {
+      throw new HttpException(
+        {
+          status: HttpStatus.INTERNAL_SERVER_ERROR,
+          error: 'Evaluation Set Not Found',
+          message: 'Failed to send evaluation error email. No evaluation set record is found for evaluation set id: ' + evalErrorParamsDTO.evaluationSetId,
+        },
+        HttpStatus.BAD_REQUEST,
+      );
+    }
+
+    let evaluationQueue: Evaluation;
+    if (evalErrorParamsDTO.evaluationId) {
+      evaluationQueue = await this.entityManager.findOneBy(Evaluation, {
+        evaluationIdentifier: evalErrorParamsDTO.evaluationId,
+      });
+    }
+
+    const evaluationStages = evalErrorParamsDTO.evaluationStages || [];
+    const rootError = !evalErrorParamsDTO.rootError ? new Error('No error message') : new Error(evalErrorParamsDTO.rootError);
+
+    await this.handleQueueingError(evaluation, evaluationQueue, evaluationStages, evaluation.userEmail, evaluation.userIdentifier, rootError, true);
+  }
+
+  async handleQueueingError(
+    evaluationSet: EvaluationSet,
+    currentEvaluationQueue: Evaluation,
+    stages: { action: string; dateTime: string }[],
+    userEmail: string,
+    userId: string,
+    rootError: Error,
+    skipUpdatingEvaluationRecords: boolean = false,
+  ) {
+    try {
+      // JSON.stringify the error note
+      const errorNote = JSON.stringify({
+        message: rootError?.message || 'No message',
+        stack: rootError?.stack || 'No stack trace available',
+        name: rootError?.name || 'UnknownError',
+      });
+
+      // Update the evaluation set status to 'ERROR'
+      try {
+        // Attempt to update evaluation queue records
+        if (!skipUpdatingEvaluationRecords) {
+          await this.evaluationSetHelper.setRecordStatusCode(
```

Comment:
This looks like the exact same call to this.evaluationSetHelper.setRecordStatusCode with the same parameters, one inside an if condition and one after that if.  This seems wrong.

Reply:
Ticket added for this issue.